project/api/write.py

```diff
@@ -153,52 +154,31 @@ def rateCivi(request):
     except Activity.DoesNotExist:
         prev_act = None
 
-    try:
+    activity_data = {
+        'account': account,
+        'thread': voted_civi.thread,
+        'civi': voted_civi,
+    }
 
-        activity_data = {
-            'account': account,
-            'thread': voted_civi.thread,
-            'civi': voted_civi,
-        }
+    activity_vote_key = 'votes_{}'.format(rating)
+    vote_val = 'vote_{}'.format(rating)
+    setattr(voted_civi, activity_vote_key, F(activity_vote_key) + 1)
```

Comment:
It resolves a problem with voting race - the previous implementation could lose some data

Reply:
Add that as a comment to the code.